packages/manager/src/components/TypeToConfirm/TypeToConfirm.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import Typography from 'src/components/core/Typography';
+import TextField from 'src/components/TextField';
+import Link from 'src/components/Link';
+import { makeStyles, Theme } from 'src/components/core/styles';
+import PreferenceToggle, { ToggleProps } from 'src/components/PreferenceToggle';
+import Grid from 'src/components/Grid';
+import CheckBox from 'src/components/CheckBox';
+import FormControlLabel from 'src/components/core/FormControlLabel';
+
+export interface Props {
+  confirmationText?: JSX.Element | string;
+  onChange: (str: string) => void;
+  label: string;
+  hideDisable?: boolean;
+  hideLabel?: boolean;
+  textFieldStyle?: Record<string, any>;
+  typographyStyle?: Record<string, any>;
+  visible?: boolean | undefined;
+  title?: string;
+  // This is a string index signature.
+  // This means that all properties in 'Props' are assignable to any
+  [propName: string]: any;
+}
+
+const useStyles = makeStyles((theme: Theme) => ({
+  description: {
+    marginTop: theme.spacing(),
+  },
+}));
+
+type CombinedProps = Props;
+
+const TypeToConfirm: React.FC<CombinedProps> = (props) => {
```

Comment:
I know this is an outstanding pattern we have, but I think its ok to do this

```suggestion
const TypeToConfirm: React.FC<Props> = (props) => {
```